cpp/src/interop/ipc.hpp

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cinttypes>
+#include <cstring>
+#include <cuda.h>
+#include <cudf/utilities/error.hpp>
+
+namespace cudf {
+namespace ipc {
+inline void check_cu_status(CUresult res)
+{
+  if (res != CUDA_SUCCESS) {
+    char const* msg;
+    if (cuGetErrorString(res, &msg) != CUDA_SUCCESS) {
```

Comment:
Just FYI, calls to the CUDA driver library will have  to be via dlopen after https://github.com/rapidsai/cudf/pull/11370.

Reply:
Thanks for the reference, will look into creating a dlopen wrapper.

Reply:
This might be a little bit trickier than I thought. With dlopen, I need to initialize the cuda context myself.